src/data/validation/email.test.js

```diff
@@ -233,4 +234,43 @@ describe('email validation', () => {
       );
     });
   });
+
+  describe('validate emails and ids extraction', () => {
+    it('extracted correct emails and ids', () => {
+      const formData = new FormData();
+      formData[EMAIL_ADDRESS_TEXT_FORM_DATA] = [
+        'abc@example.com,000000000000ABCABC',
+        'asdf@example.com,',
+        'zzz@example.com,000000000000XYZXYZ',
+      ].join('\n');
+      formData[EMAIL_ADDRESS_CSV_FORM_DATA] = [
+        'one@example.com,000000000000YYYYYY',
+        'two@example.com,000000000000ZZZZZZ',
+        'three@example.com,000000000000ABCDDD',
+        'wow@example.com,',
```

Comment:
Nit: consider adding one more test email in here without a trailing comma, like "nocomma@example.com" to make sure the splitting/parsing still works as expected.